pecos/core/utils/newton.hpp

```diff
@@ -272,5 +272,120 @@ namespace pecos {
             return cg_iter;
         };
     };
+
+
+    // Platt scale with given target curve.
+    // Reference Implementation:
+    //  https://github.com/cjlin1/libsvm/blob/master/svm.cpp
+
+    #define Malloc(type,n) (type *)malloc((n)*sizeof(type))
```

Comment:
I suggest if we do not directly copy from 3rd party code, we should use container maintained memory space like vector instead of malloc/free, but it is up to you.